libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -360,8 +381,8 @@ fn notify_subscribers(
     });
 }
 
-fn send_heartbeats(
-    state: &mut ConnectionState,
+fn send_heartbeats<T: MatrixLifeCycle, U: MatrixSender>(
+    state: &mut ConnectionState<T, U>,
```

Comment:
Given that this is private and only uses the `MatrixSender`, you can change the parameters to take a `&dyn MatrixSender` and you won't need the generics or the whole state

Reply:
It would still need state to access connection metadata though..

Reply:
Ah, I didn't see the later use of state